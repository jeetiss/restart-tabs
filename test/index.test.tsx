/* eslint-env jest */

import "@testing-library/jest-dom/extend-expect";

import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Panel, Tab, Tabs } from "../src";

const Button: React.FC<{ isActive?: boolean; onClick?: () => void }> = ({
  isActive,
  onClick,
  children,
}) => (
  <button disabled={isActive} onClick={onClick}>
    {children}
  </button>
);

const Testing = () => (
  <Tabs>
    <div>
      <Tab>
        <Button>tab 1</Button>
      </Tab>
      <Tab>
        <Button>tab 2</Button>
      </Tab>
    </div>

    <Panel data-testid="panel-1">content 1</Panel>
    <Panel>content 2</Panel>
  </Tabs>
);

describe("Tabs", () => {
  test("should display same panel on clicking same tab", () => {
    const { container, queryByText } = render(<Testing />);

    expect(container).toHaveTextContent("content 1");
    expect(container).not.toHaveTextContent("content 2");

    fireEvent.click(queryByText("tab 1")!);

    expect(container).toHaveTextContent("content 1");
    expect(container).not.toHaveTextContent("content 2");
  });

  test("should display correct tab on change", () => {
    const { container, queryByText } = render(<Testing />);

    expect(container).toHaveTextContent("content 1");
    expect(container).not.toHaveTextContent("content 2");

    fireEvent.click(queryByText("tab 2")!);

    expect(container).not.toHaveTextContent("content 1");
    expect(container).toHaveTextContent("content 2");
  });
});
