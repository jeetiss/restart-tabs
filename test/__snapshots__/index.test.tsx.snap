// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      disabled=""
    >
      tab 1
    </button>
    <button>
      tab 2
    </button>
  </div>
  content 1
</DocumentFragment>
`;
